Source/WebCore/rendering/svg/SVGTextQuery.cpp

```diff
@@ -155,82 +155,45 @@ bool SVGTextQuery::mapStartEndPositionsIntoFragmentCoordinates(Data* queryData,
-    for (; textMetricsOffset < textMetricsSize && positionOffset < positionSize; ++textMetricsOffset) {
+    // Compute the offset of the fragment within the box, since that's the
+    // space <startPosition, endPosition> is in (and that's what we need).
+    unsigned fragmentOffsetInBox = fragment.characterOffset - queryData->textBox->start();
+    unsigned fragmentEndInBox = fragmentOffsetInBox + fragment.length;
+    // Find the text metrics cell that start at or contain the character startPosition.
+    while (fragmentOffsetInBox < fragmentEndInBox) {
         SVGTextMetrics& metrics = textMetricsValues[textMetricsOffset];
-
-        // Advance to text box start location.
-        if (positionOffset < boxStart) {
-            positionOffset += metrics.length();
-            continue;
-        }
-
-        // Stop if we've finished processing this text box.
-        if (positionOffset >= boxStart + boxLength)
+        int glyphEnd = fragmentOffsetInBox + metrics.length();
+        if (startPosition < glyphEnd)
             break;
-
-        // If the start position maps to a character in the metrics list, we don't need to modify it.
-        if (startPosition == positionOffset)
-            alterStartPosition = false;
-
-        // If the start position maps to a character in the metrics list, we don't need to modify it.
-        if (endPosition == positionOffset)
-            alterEndPosition = false;
-
-        // Detect ligatures.
-        if (lastPositionOffset && lastPositionOffset.value() - positionOffset > 1) {
-            if (alterStartPosition && startPosition > lastPositionOffset.value() && startPosition < positionOffset) {
-                startPosition = lastPositionOffset.value();
-                alterStartPosition = false;
-            }
-
-            if (alterEndPosition && endPosition > lastPositionOffset.value() && endPosition < positionOffset) {
-                endPosition = positionOffset;
-                alterEndPosition = false;
-            }
-        }
-
-        if (!alterStartPosition && !alterEndPosition)
-            break;
-
-        lastPositionOffset = positionOffset;
-        positionOffset += metrics.length();
+        fragmentOffsetInBox = glyphEnd;
+        textMetricsOffset++;
     }
+    startPosition = fragmentOffsetInBox;
 
-    if (!alterStartPosition && !alterEndPosition)
-        return;
-
-    if (lastPositionOffset && lastPositionOffset.value() - positionOffset > 1) {
-        if (alterStartPosition && startPosition > lastPositionOffset.value() && startPosition < positionOffset)
-            startPosition = lastPositionOffset.value();
-
-        if (alterEndPosition && endPosition > lastPositionOffset.value() && endPosition < positionOffset)
-            endPosition = positionOffset;
+    // Find the text metrics cell that contain or ends at the character endPosition.
+    while (fragmentOffsetInBox < fragmentEndInBox) {
+        SVGTextMetrics& metrics = textMetricsValues[textMetricsOffset];
```

Comment:
nit: `auto&`

Reply:
Done in [9659667](https://github.com/WebKit/WebKit/commit/965966790e575c173a0387d457847007605d0525).